src/material/timepicker/timepicker.md

```diff
@@ -1 +1,310 @@
-TODO
+The Angular Material Timepicker allows users to set the time portion of a date object either by
+typing it in or by selecting it from a list of pre-defined options.
+
+<!-- example(timepicker-overview) -->
+
+### Connecting a timepicker to an input
+
+A timepicker is composed of a text input and a dropdown menu, connected through the `matTimepicker`
+binding on the input.
+
+There is also an optional timepicker toggle button that gives the user an easy way to open the dropdown.
+
+<!-- example({"example":"timepicker-overview",
+              "file":"timepicker-overview-example.html",
+              "region":"minimum-setup"}) -->
+
+The timepicker input and toggle can be used either on their own or as a part of a `mat-form-field`:
+
+<!-- example({"example":"timepicker-overview",
+              "file":"timepicker-overview-example.html"}) -->
+
+### Timepicker forms integration
+
+The timepicker input integrates with the `@angular/forms` module by providing itself as a
+`ControlValueAccessor` and a `Validator` (see [Input validation](#input-validation) below for more
+information). When the user types in a new time or selects one from the dropdown, the
+time will be set on the date object which is the current value of the form control. If the form
+control doesn't have a value, the timepicker will create one with today's date and the selected
+time.
+
+<!-- example(timepicker-forms) -->
+
+
+### Integration with `MatDatepicker`
+
+Material's datepicker and timepicker components can operate over the same value, allowing for a
+combined datetime picker to be implemented. When binding the two components to the same value, the
+datepicker will set the entire date object while the timepicker will only modify the time portion
+of it.
+
+<!-- example(timepicker-datepicker-integration) -->
+
+### Input validation
+
+The timepicker input checks that the value typed in by the user is a valid time string and
+whether it fits into the specified bounds.
+
+If the user types in an invalid time string (for example `abc` or `24:67`), the timepicker input
+will report the `matTimepickerParse` error. The string is parsed using the `parseTime` method of
+the [the current date implementation](#choosing-a-date-implementation-and-format-settings).
```

Comment:
Maybe the choosing a date implementation and format section should appear before this section does?